core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -213,38 +213,24 @@ public BlockReader getBlockReader(long sessionId, long blockId, long lockId)
   public TempBlockMeta createBlock(long sessionId, long blockId, BlockStoreLocation location,
       long initialBlockSize)
           throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException {
-    if (RESERVER_ENABLED) {
```

Comment:
Also drop the code path for `RESERVER_ENABLED`, because the block might never be evicted and the retry times out if the high watermark is configured not to be enough to trigger and free the requested amount.

Reply:
This will cause sync eviction to be triggered along with async eviction, which is not what we want. In the future, we can leverage the fact that only async eviction occurs to gather additional metadata which would otherwise be too expensive to do on every sync eviction.

Reply:
will this change be safe in case users change the reserver option from default `true` to be `false`